### PR TITLE
add a function to retrieve a hvdc line and related shunt compensators

### PR DESCRIPTION
**Pull request by dbraquart**

### CODE REVIEW

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java [hunk 1]
```diff
@@ -323,6 +327,31 @@ public ElementInfos getBranchOrThreeWindingsTransformer(UUID networkUuid, String
         throw new ResponseStatusException(HttpStatus.NO_CONTENT);
     }
 
+    private static List<SelectedShuntCompensatorData> toShuntCompensatorData(String lccBusOrBusbarSectionId, Stream<ShuntCompensator> shuntCompensators) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/NetworkMapService.java:330]: toShuntCompensatorInfos

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java [hunk 2]
```diff
@@ -323,6 +327,31 @@ public ElementInfos getBranchOrThreeWindingsTransformer(UUID networkUuid, String
...
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    public HvdcShuntCompensatorInfos getHvcLineWithShuntCompensators(UUID networkUuid, String variantId, String hvdcId) {
+        HvdcShuntCompensatorInfos.HvdcShuntCompensatorInfosBuilder builder = HvdcShuntCompensatorInfos.builder().id(hvdcId);
+        Network network = getNetwork(networkUuid, PreloadingStrategy.NONE, variantId);
+        HvdcLine hvdcLine = network.getHvdcLine(hvdcId);
+        if (hvdcLine != null) { // it could be OK to not find the hvdc: can be used in a modification with un-existing hvdc
+            builder.hvdcType(hvdcLine.getConverterStation1().getHvdcType());
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/NetworkMapService.java:344]: see remark above on hvdcType field utility ...

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java [hunk 3]
```diff
@@ -323,6 +327,31 @@ public ElementInfos getBranchOrThreeWindingsTransformer(UUID networkUuid, String
...
+                .map(s -> SelectedShuntCompensatorData.builder()
+                        .id(s.getId())
+                        .selected(Objects.equals(lccBusOrBusbarSectionId, getBusOrBusbarSection(s.getTerminal())))
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    public HvdcShuntCompensatorInfos getHvcLineWithShuntCompensators(UUID networkUuid, String variantId, String hvdcId) {
+        HvdcShuntCompensatorInfos.HvdcShuntCompensatorInfosBuilder builder = HvdcShuntCompensatorInfos.builder().id(hvdcId);
+        Network network = getNetwork(networkUuid, PreloadingStrategy.NONE, variantId);
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/NetworkMapService.java:341]: Maybe PreloadingStrategy.COLLECTION, as we can retrieve also multiple shunt compensators ...
But it's not very important ...

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

**Comment by SlimaneAmar** [comment]
Why we do not send only selected shunt compensators from front ?

File: src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+@NoArgsConstructor
+@Getter
+@Setter
+@ToString(callSuper = true)
+@Schema(description = "HVDC deletion")
+public class HvdcShuntCompensatorInfos {
+
+    private String id;
+
+    private HvdcConverterStation.HvdcType hvdcType;
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java:41]: yes maybe I dont - I will check

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java (hunk 3 above)
Comment (dbraquart) [reply src/main/java/org/gridsuite/network/map/NetworkMapService.java:328]: not sure it worths it - to be checked

File: src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * @author David Braquart <david.braquart at rte-france.com>
+ */
+@SuperBuilder
+@NoArgsConstructor
+@Getter
+@Setter
+@ToString(callSuper = true)
+@Schema(description = "HVDC deletion")
+public class HvdcShuntCompensatorInfos {
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java:37]: HvdcShuntCompensatorInfos is good, isnt it ?

File: src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java:41]: done

**Review by FranckLecuyer — APPROVED** [review]

File: src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import static org.gridsuite.network.map.dto.utils.ElementUtils.getBusOrBusbarSection;
+
+/**
+ * @author David Braquart <david.braquart at rte-france.com>
+ */
+@SuperBuilder
+@NoArgsConstructor
+@Getter
+@Setter
+@ToString(callSuper = true)
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java:32]: Remove `callSuper`

File: src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+@Schema(description = "HVDC deletion")
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java:33]: ?

File: src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java:34]: Oh sorry
By HvdcShuntCompensator**s**Infos

File: src/main/java/org/gridsuite/network/map/dto/hvdc/SelectedShuntCompensatorData.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * @author David Braquart <david.braquart at rte-france.com>
+ */
+@Builder
+@Getter
+@EqualsAndHashCode
+public class SelectedShuntCompensatorData {
+    private String id;
+
+    private boolean selected;
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/network/map/dto/hvdc/SelectedShuntCompensatorData.java]: already changed with last commits

File: src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java (hunk 7 above)
Comment (dbraquart) [reply src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorInfos.java:33]: removed, no need

File: src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorsInfos.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+ */
+@SuperBuilder
+@Getter
+@Setter
+public class HvdcShuntCompensatorsInfos {
+
+    @Builder
+    @Getter
+    @EqualsAndHashCode
+    static class ShuntCompensatorInfos {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/dto/hvdc/HvdcShuntCompensatorsInfos.java:35]: Add private

File: src/main/java/org/gridsuite/network/map/NetworkMapController.java [hunk 10]
```diff
@@ -128,12 +129,24 @@ public class NetworkMapController {
...
         return networkMapService.getBranchOrThreeWindingsTransformer(networkUuid, variantId, equipmentId);
     }
+
+    @GetMapping(value = "/networks/{networkUuid}/hvdc-lines/{hvdcId}/shunt-compensators", produces = APPLICATION_JSON_VALUE)
+    @Operation(summary = "For a given HVDC line, get its related shunt compensators in case of LCC converter station")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Hvdc line type and its shunt compensators on each side"),
+        @ApiResponse(responseCode = "204", description = "No element found")
+    })
+    public @ResponseBody HvdcShuntCompensatorsInfos getHvcLineWithShuntCompensators(@Parameter(description = "Network UUID") @PathVariable("networkUuid") UUID networkUuid,
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/NetworkMapController.java:147]: Rename by `getHvcLineShuntCompensators` ?

File: src/main/java/org/gridsuite/network/map/NetworkMapController.java [hunk 11]
```diff
@@ -128,12 +129,24 @@ public class NetworkMapController {
...
+                                                                                    @Parameter(description = "Hcdc Line ID") @PathVariable("hvdcId") String hvdcId,
+                                                                                    @Parameter(description = "Variant ID") @RequestParam(name = "variantId", required = false) String variantId) {
+        return networkMapService.getHvcLineWithShuntCompensators(networkUuid, variantId, hvdcId);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/NetworkMapController.java:150]: Same

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**